core/blockly.js

```diff
@@ -450,10 +450,14 @@ Blockly.bindEventWithChecks_ = function(node, name, thisObject, func,
     }
   };
 
-  node.addEventListener(name, wrapFunc, false);
-  var bindData = [[node, name, wrapFunc]];
+  var bindData = [];
+  // Don't register the mouse event if an equivalent pointer event is supported.
+  if (!window.PointerEvent || !(name in Blockly.Touch.TOUCH_MAP)) {
```

Comment:
window might not exist in headless mode. Are we certain this function will never be called in headless? I feel more comfortable checking `window` too.

Reply:
Good point.  That also means that there should be a check for window in touch.js: 
```
Blockly.Touch.TOUCH_MAP = {};
if (window.PointerEvent) {
  Blockly.Touch.TOUCH_MAP = {
    'mousedown': ['pointerdown'],
    'mousemove': ['pointermove'],
    'mouseup': ['pointerup', 'pointercancel']
  };
} else if (goog.events.BrowserFeature.TOUCH_ENABLED) {
  Blockly.Touch.TOUCH_MAP = {
    'mousedown': ['touchstart'],
    'mousemove': ['touchmove'],
    'mouseup': ['touchend', 'touchcancel']
  };
}

```

In the particular case of `bindEvent_`, it's called from inject.js with `window` as the `node` argument, so we may have already failed.

It looks like we used to create a fake window object to solve this problem (not the best solution): https://github.com/google/blockly/pull/1566/files  Now that's gone.